redbox-review/redbox.go

```diff
@@ -0,0 +1,293 @@
+package redbox
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cgclever/redbox/s3box"
+)
+
+const defaultNManifests = 4
+
+var (
+	errShippingInProgress = fmt.Errorf("Cannot perform any action when shipping is in progress.")
+	errIncompleteArgs     = fmt.Errorf("Creating a redshift box requires a schema, table and an s3 bucket.")
+	errInvalidJSONInput   = fmt.Errorf("Only JSON inputs are supported.")
+	errBoxShipped         = fmt.Errorf("Cannot perform any actions, the box has been shipped.")
+	errNothingToShip      = fmt.Errorf("Cannot perform send, no data was packed.")
+)
+
+// Redbox manages piping data into Redshift.
+// An S3Box is used to manage data transport to S3 via Pack, after
+// which Ship commits the data to Redshift.
+type Redbox struct {
```

Comment:
yeah, i like that model. I like the idea of keep the public interface (e.g. the struct) cleaner if possible and doing `schema := rb.o.Schema` seems like a good way to do that